corehq/apps/userreports/views.py

```diff
@@ -483,27 +483,43 @@ def page_context(self):
     def form(self):
         max_allowed_reports = allowed_report_builder_reports(self.request)
         if self.request.method == 'POST':
-            return DataSourceForm(self.domain, self.report_type, max_allowed_reports, self.request.POST)
-        return DataSourceForm(self.domain, self.report_type, max_allowed_reports)
+            return DataSourceForm(self.domain, max_allowed_reports, self.request.POST)
+        return DataSourceForm(self.domain, max_allowed_reports)
+
+    def _get_config_kwargs(self, app_source):
+        app = Application.get(app_source.application)
+        builder = DataSourceBuilder(self.domain, app, app_source.source_type, app_source.source)
+        return {
+            'display_name': builder.data_source_name,
+            'referenced_doc_type': builder.source_doc_type,
+            'configured_filter': builder.filter,
+            'configured_indicators': builder.indicators(),
+            'base_item_expression': builder.base_item_expression(False),
+            'meta': DataSourceMeta(
+                build=DataSourceBuildInformation(
+                    source_id=app_source.source,
+                    app_id=app._id,
+                    app_version=app.version,
```

Comment:
Not pressing, but it just came to mind that it may be useful to also add some sort flag to the data source meta data that identifies this data source as a temporary report builder preview data source. That meta property might be used to filter these out of the side bar on the `/a/<domain>/configurable_reports` pages for instance.